### PR TITLE
Fix Building Toolchain Installer

### PR DESCRIPTION
**Pull request by leon-anavi**
These patches fix building toolchain installer by updating RVI SOTA client and meta-rust (in genivi-dev-platform). Details are available at JIRA issue GDP-539.

Best regards,
Leon

### CODE REVIEW

File: meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bb [hunk 1]
```diff
@@ -112,10 +112,9 @@ SRC_URI[index.sha256sum] = "86114b93f1f51aaf0aec3af0751d214b351f4ff9839ba031315c
 
 SYSTEMD_SERVICE_${PN} = "sota_client.service"
 
-DEPENDS += " openssl "
+DEPENDS += " openssl dbus "
```
Comment (zeenix) [inline meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bb:115]: The commit log mentions the end result but now why the change is needed (for build to work).

Comment (leon-anavi) [reply meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bb:115]: RVI SOTA client is written in the Rust programming language therefore it depends on Yocto/OE layer meta-rust which provides Rust and Cargo. Crate dbus is required by RVI SOTA Client and it is related to the Yocto/OE recipe for dbus. While I was working on the update of meta-rust and RVI SOTA Client, I noticed that in some cases the build may fail if dbus is not present at DEPENDS therefore I moved it from RDEPENDS to DEPENDS. 

Best regards,
Leon

Comment (zeenix) [reply meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bb:115]: Thanks for all the details Leon but my question still remain unanswered, why is this needed now? Did they now start building against dbus library or it is just that we only noticed this now by coincidence?

Comment (zeenix) [reply meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bb:115]: and we'd want that in the commit log.

**Review by zeenix — COMMENTED** [review]
I think the missing comma in shorlog makes it sound like you are updating openssl:

"rvi-sota-client: Update, & fix openssl" would be better.

Although I don't think you're fixing openssl here, so I'd just write:

"rvi-sota-client: Update, & remove a workaround"

Patch itself looks fine.

**Review by zeenix — COMMENTED** [review]
I don't think this is a good idea. meta-rust providing libgit2 is a side-effect. We should be able to remove meta-rust layer entirely without any breakage if all dependencies stop using rust (Sota client might be doing so in near future).

File: meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bb (hunk 1 above)
Comment (leon-anavi) [reply meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bb:115]: Sorry, I do not understand what is not clear. I will try to answer to your questions and provide more details.
The change is needed because otherwise in certain cases the build of RVI SOTA Client may fail which will lead to failure of building the whole GDP image. I noticed this issue recently. For long time dbus existed as a run-time dependency (RDEPENDS) and now I have just moved it to DEPENDS. In my opinion it is a race condition because if something else bitbakes dbus before building RVI SOTA client the issue will not occur.

**Comment by zeenix** [comment]
oh great my last comments were taken out of context, thanks github! I'm sure one of them is obvious but the last one is about removal of libgit2 patch.

File: meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bb (hunk 1 above)
Comment (zeenix) [reply meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bb:115]: Unclear part: Why this fix works? More precisely, what is the "exact" issue this resolves. At the expanse of repeating myself: "Did they now start building against dbus library or it is just that we only noticed this now by coincidence?"

Comment (leon-anavi) [reply meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bb:115]: As I said, I noticed the issue recently while I was testing the bunch of patches. I have not investigated if and how often was it appearing in older versions of RVI SOTA Client.

Best regards, Leon

**Comment by leon-anavi** [comment]
The removal of libgit2 has been already discussed in the mailing list. Apart from RVI SOTA Client nothing else depends on libgit2. I have added the recipe to meta-genivi-dev on 13 December because it was then needed for the update of RVI SOTA Client. Now the same recipe is provided through meta-rust therefore it would be better to remove it from meta-genivi-dev. As of the moment RVI SOTA clients depends on meta-rust and it cannot be built without this layer. 

Best regards,
Leon

File: meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bb (hunk 1 above)
Comment (zeenix) [reply meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bb:115]: OK, but I'm sure you can understand that we can't call something a fix without knowing how/why it works cause it could just be a race-condition/coincidence that patch *seems* to work and in reality it doesn't. That's why we need to know the specific reason. You've done most of the work already and mostly likely your patch is good so I'm hoping that you'll have time to find out and be able to provide a complete story in commit log.

**Comment by zeenix** [comment]
> Apart from RVI SOTA Client nothing else depends on libgit2. 

Ah OK, didn't know that. Then it's fine but you should mention in the commit log that very important fact.

File: meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bb (hunk 1 above)
Comment (zeenix) [reply meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bb:115]: I did a bit of digging and seems dbus as build-dep was added in 2015: https://github.com/advancedtelematic/rvi_sota_client/blob/master/Cargo.toml#L21

So it's most likely that we have been just lucky so far that dbus was built before sota-client. could you please update the commit logs?

Comment (leon-anavi) [reply meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bb:115]: @zeenix OK, thank your for the feedback. @chbae has already merged the GitHub pull request with these patches.

Best regards,
Leon

Comment (zeenix) [reply meta-genivi-dev/recipes-sota/rvi-sota-client/rvi-sota-client_git.bb:115]: oh, i missed that fact somehow. :( I'll use the "first day after vacation" excuse again. :) @chbae thanks for the quick action.

**Comment by chbae** [comment]
I already reviewed this changes and commit messages :).
if possible, we can make a decision about commit message rule in GDP call or mailing.

Commit message is also really important.

**Comment by zeenix** [comment]
> I already reviewed this changes and commit messages :).

Thanks.

> if possible, we can make a decision about commit message rule in GDP call or mailing.

We already did some weeks back. I guess you missed that one.
